src/server/game/map/tile/city.ts

```diff
@@ -59,4 +74,283 @@ export class City {
   removeTile(coords: Coords) {
     this.tiles.delete(coords);
   }
+
+  getUnits(): Unit[] {
+    return this.units;
+  }
+
+  getUnitPositions(): Coords[] {
+    return this.units.map(unit => unit.coords);
+  }
+
+  addUnit(unit: Unit): void {
+    this.units.push(unit);
+    if (this instanceof BarbarianCamp) {
+      unit.setBarbarian(true);
+    }
+  }
+
+  removeUnit(unit: Unit): void {
+    const unitIndex = this.units.indexOf(unit);
+    if (unitIndex > -1) {
+      this.units.splice(unitIndex, 1);
+    }
+  }
+
+  turn(world: World, map: Map): void {
+    // By default, do nothing (for now)
+  }
+
+
+}
+
+export class UnitController extends City {
+  private world: World;
+  private map: Map;
+
+  moveUnit(unit: Unit, toPos: Coords, allowCombat?: boolean): boolean {
+    const tile = this.map.getTile(toPos);
+    if (!tile) return false;
+    const movementCost = tile.getMovementCost(unit, getDirection(toPos, unit.coords));
+    if (!(unit.movement < movementCost)) {
```

Comment:
I would change this to an early return to make this more readable. (Yes, I'm annoying)

Reply:
@kjniemela This is probably the most important one.

Reply:
That's what we do these reviews for 👍